packages/p2p-v2/src/pages/my-profile/screens/MyProfileCounterparties/MyProfileCounterparties.tsx

```diff
@@ -0,0 +1,109 @@
+import MyProfileCounterpartiesHeader from './MyProfileCounterpartiesHeader';
+import MyProfileCounterpartiesTable from './MyProfileCounterpartiesTable';
+import MyProfileCounterpartiesTableRow from './MyProfileCounterpartiesTableRow';
+import './MyProfileCounterparties.scss';
+
+type TMyProfileCounterpartiesTableRowRendererProps = {
+    id?: string | undefined;
+    is_blocked: boolean;
+    name?: string | undefined;
+};
+
+const MyProfileCounterpartiesDisplayWrapper = ({ children }: PropsWithChildren<unknown>) => {
+    const { isMobile } = useDevice();
+    if (isMobile) {
+        return (
+            <FullPageMobileWrapper
+                noFooter
+                renderHeader={() => (
+                    <Text className='p2p-v2-my-profile-counterparties__header' size='md' weight='bold'>
+                        My counterparties
+                    </Text>
+                )}
+            >
+                {children}
+            </FullPageMobileWrapper>
+        );
+    }
+    return children;
+};
+
+const MyProfileCounterpartiesTableRowRenderer = ({
+    id,
+    is_blocked,
+    name,
+}: TMyProfileCounterpartiesTableRowRendererProps) => (
+    <MyProfileCounterpartiesTableRow id={id!} isBlocked={is_blocked} nickname={name!} />
+);
+
+const MyProfileCounterparties = () => {
+    const [searchValue, setSearchValue] = useState('');
+    const [dropdownValue, setDropdownValue] = useState('all');
+    const [isFilterModalOpen, setIsFilterModalOpen] = useState(false);
+
+    const {
+        data = [],
+        isFetching,
+        isLoading,
+        loadMoreAdvertisers,
+    } = p2p.advertiser.useGetList({
+        trade_partners: 1,
+        is_blocked: dropdownValue === 'blocked' ? 1 : 0,
+        advertiser_name: searchValue,
+    });
+
+    const onClickFilter = () => {
+        setIsFilterModalOpen(true);
+    };
+
+    const onToggle = (value: string) => {
+        setDropdownValue(value);
+        setIsFilterModalOpen(false);
+    };
+
+    return (
+        <MyProfileCounterpartiesDisplayWrapper>
+            <div className='p2p-v2-my-profile-counterparties'>
+                <div>
+                    <Text as='p' size='sm'>
+                        {`When you block someone, you won’t see their ads, and they can’t see yours. Your ads will be hidden from their search results, too.`}
+                    </Text>
+                    <MyProfileCounterpartiesHeader
+                        dropdownValue={dropdownValue}
+                        onClickFilter={onClickFilter}
+                        setDropdownValue={setDropdownValue}
+                        setSearchValue={setSearchValue}
+                    />
+                </div>
+                {!isFetching && data.length === 0 && searchValue === '' ? (
+                    <MyProfileCounterpartiesEmpty />
+                ) : (
+                    <div className='p2p-v2-my-profile-counterparties__content'>
+                        <MyProfileCounterpartiesTable
```

Comment:
Can we move the hook call `p2p.advertiser.useGetList` to the component `MyProfileCounterpartiesTable` itself? The query will be cached so there won't be unecessary recalls

Reply:
Then we can get rid of `data`, `isFetching`, `isLoading`